rsocket/src/transport/socket.rs

```diff
@@ -352,8 +362,11 @@ impl DuplexSocket {
                     Handler::ReqRC(sender) => {
                         // TODO: support channel
                         if flag & Frame::FLAG_NEXT != 0 {
-                            if let Err(_) = sender.clone().send(Ok(input)).await {
+                            if sender.is_closed() {
```

Comment:
Unfortunately this happens only after the server sends the next frame.
I will try to implement the drop() function for a custom stream wrapper which will send the CANCEL frame immediately.

Reply:
👍